config/cmd/cmd.go

```diff
@@ -570,6 +575,63 @@ func (c *cmd) Before(ctx *cli.Context) error {
 	microClient := wrapper.CacheClient(cacheFn, grpc.NewClient())
 	microClient = wrapper.AuthClient(authFn, microClient)
 
+	// Setup auth options
+	authOpts := []auth.Option{auth.WithClient(microClient)}
+	if len(ctx.String("auth_address")) > 0 {
+		authOpts = append(authOpts, auth.Addrs(ctx.String("auth_address")))
+	}
+	if len(ctx.String("auth_id")) > 0 || len(ctx.String("auth_secret")) > 0 {
```

Comment:
Should this be an `&&` rather than an `||`?

Reply:
No, if either value is set we want to set the option. If it's incorrect the auth service will be responsible for returning an error on startup. We only really have the check so we don't call Init without reason.